courses/javascript/classes.md

```diff
@@ -0,0 +1,217 @@
+
+```javascript
+class TestClass {
+
+    constructor(){
+        console.log("constructor is called");
+    }
+}
+
+let textClassObjects = new TestClass();
+
+```
+
+The constructor will be called when you created the object named **textClassObjects**.
+
+You can also pass data through the constructor just like a function.
+
+```javascript
+
+class TestClass {
+
+    constructor(name){
+        console.log(`this is the passes data ${name}`);
+    }
+}
+
+let textClassObjects = new TestClass("nullcast");
+
+```
+* Please note that you can only create one constructor in a class.
+***
+### this keyword
+In JavaScript this keyword refers to the object it belongs to.
+
+We will have some examples of **this** in **classes**.
+
+```javascript
+
+class TestClass {
+
+    name = 'nullcast'
+
+    constructor(name){
+        
+        // this is the same one from class
+        console.log(this.name);
+
+        // the name is from the params
+        console.log('data from params '+ name); 
+    }
+}
+
+let textClassObjects = new TestClass('Ducks');
+
+```
+In the above example, 'this.name' is used to access the name variable on the class named 'TestClass'. And the 'name' variable is the same variable we need as params in our constructor.
+
+* Please note that they are actually called methods not variables. But for the simple understanding we just called them variables.
+***
+## Methods
+Just like we use variables and constructor in our class. We can also create functions too. We called then methods. They looks like a function and act like one for the most part.
+
+For example:
+
+```javascript
+
+class TestClass {
+
+    constructor(name){
+    
+        this.stringOne = name
+    }
+
+    toConcatStrings(stringTwo) {
+
+        return this.stringOne + stringTwo
+
+    }
+    
+}
+
+let textClassObjects = new TestClass('Nullcast');
+
+const concatData = textClassObjects.toConcatStrings(' Ducks')
+
+// result: Nullcast Ducks
+ console.log(concatData); 
+
+```
+
+In the above example, we create a class and it has a method named toConcatStrings. In the constructor of that class, you can see we have a param named name and we initialize that value to this.stringOne. But if you take look at the class you can see we didn't declare any variables or method named stringOne. Don't worry it will not cause an error a new method named stringOne will be created for our class.
+
+You can see we called our method outside of our class by `const concatData = textClassObjects.toConcatStrings(' Ducks')`. By this, we can point out that we can call the method named **toConcatStrings** on the class **textClassObjects**.
+
+## Complete the task below
+- create a class named Calcu
+- create a constructor with params as num1
+- make an object name num1 using this keyword and insert the above num1 param to it.
+- make a method name toAdd which have a param named num2
```

Comment:
~name~ named, toAdd as inline code block, ~'which have a'~ Which accepts a, num2 in inline code block